### DISCUSSION TITLE
Support for zip files

### DISCUSSION THREAD

**Discussion by fstern** [jonaswinkler/paperless-ng#596] [Feature Requests]
Hi, 

Some companies (e.g. Deutsche Telekom) send their invoices as a ZIP file, containing a PDF file and a pkcs12 signature. 

Would it be possible for the paperless-ng worker to extract ZIP files, add the PDF to the library and remove any other files from the consume directory?

**Comment by jonaswinkler** [comment]
Given the current architecture of paperless, the following would be possible;

* Allow paperless to detect and process zip files.
* Extract any given zip file, and submit all supported documents to the consumer. This is similar to putting the files in the consumption directory; but rather than doing that, I'd tell paperless to add the extracted documents directly, skipping the consumption directory.
* The zip file will get deleted if it was put in the consumption directory before any of the other documents finish.

I won't be able to just write something that extracts zip files in the consumption directory in place, since I'd also like to support zip files sent via mail, or uploaded on the front end.

**Reply by fstern** [reply]
This sounds perfect for my usecase - so paperless could even pick up documents directly from my mailbox.

**Comment by Boldie** [comment]
I have a quite different need: I have some medical data which I want to store beside the rest of my healt documents. So I want to upload them as zip file (since they are a lot of files, but there is no need to interprete them by paperless) and keep them inside of paperless.

**Reply by jonaswinkler** [reply]
Well, paperless is for documents with text on them, and not a general purpose tool for arbitrary data.

**Reply by Boldie** [reply]
That is true, but why not? Why should I have one archive for paper documents and another storage for the other data, especially if I want to archive them? Archive means for me: I have to keep them, but it is very unlikely to need them for the moment or in the future. One of these examples is for me also my tax files for preparing my yearly tax stuff. I add the documents sent to the tax agency to paperless and the rest?
From my opinion of paperless also means other stuff must be paperless and without any breaks in the workflow; I want to have a single point of responsible or at least a point I can start over if I think of my archive system.
